code/modules/mob/living/basic/lavaland/legion/legion_tumour.dm

```diff
@@ -0,0 +1,124 @@
+/// Left behind when a legion infects you, for medical enrichment
+/obj/item/organ/internal/legion_tumour
+	name = "legion tumour"
+	desc = "A mass of pulsing flesh and dark tendrils."
+	icon = 'icons/obj/medical/organs/mining_organs.dmi'
+	icon_state = "legion_remains"
+	zone = BODY_ZONE_CHEST
+	slot = ORGAN_SLOT_PARASITE_EGG
+	/// What stage of growth the corruption has reached.
+	var/stage = 0
+	/// We apply this status effect periodically or when used on someone
+	var/applied_status = /datum/status_effect/regenerative_core
+	/// How long have we been in this stage?
+	var/elapsed_time = 0 SECONDS
+	/// How long does it take to advance one stage?
+	var/growth_time = 80 SECONDS // Long enough that if you go back to lavaland without realising it you're not totally fucked
+	/// What kind of mob will we transform into?
+	var/spawn_type = /mob/living/basic/mining/legion
+	/// Spooky sounds to play as you start to turn
+	var/static/list/spooky_sounds = list(
+		'sound/voice/lowHiss1.ogg',
+		'sound/voice/lowHiss2.ogg',
+		'sound/voice/lowHiss3.ogg',
+		'sound/voice/lowHiss4.ogg',
+	)
+
+/obj/item/organ/internal/legion_tumour/Remove(mob/living/carbon/egg_owner, special)
+	. = ..()
+	stage = 0
+	elapsed_time = 0
+
+/obj/item/organ/internal/legion_tumour/attack(mob/living/target, mob/living/user, params)
+	if (try_apply(target, user))
+		qdel(src)
+		return
+	return ..()
+
+/// Smear it on someone like a regen core, why not
+/obj/item/organ/internal/legion_tumour/proc/try_apply(mob/living/target, mob/user)
+	if(!user.Adjacent(target) || !isliving(target) || target.stat == DEAD)
+		return FALSE
+
+	target.apply_status_effect(applied_status)
+	target.add_mood_event(MOOD_CATEGORY_LEGION_CORE, /datum/mood_event/healsbadman)
```

Comment:
If I'm reading this right, it means that legion corpses have a 25% chance to have what is effectively a medium-sized regenerative core that doesn't decay. It won't zombify the body if they're not dead, which I dunno if it's intentional or not.

Cool, Ashwalkers will love this I think.

Did you make lings purge the organ with Anatomic Panacea?

Reply:
If you bother to disembowel it out of someone I guess you get a regenerative core which doesn't decay yeah. Maybe it _should_ create a legion if used directly on a corpse which would be kind of neat.
Good catch with Panacea I forgot about it.